### PR TITLE
documents: replace files editor

### PR DESCRIPTION
**Pull request by jma**
image

### CODE REVIEW

File: projects/sonar/src/app/record/files/file-item/file-item.component.html [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    <div class="col">
+      <div class="flex justify-content-end flex-wrap">
+        <p-fileUpload
+          mode="basic"
+          [chooseLabel]="'A new version' | translate"
+          chooseIcon="pi pi-upload"
+          [accept]="file().metadata.mimetype"
+          fileLimit="1"
+          [auto]="true"
+          [maxFileSize]="500 * 1024 * 1024"
```
Comment (Garfield-fr) [inline projects/sonar/src/app/record/files/file-item/file-item.component.html]: Why did you hardcode the value ?

File: projects/sonar/src/app/record/files/upload-files/upload-files.component.html [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  name="file-upload"
+  color="#fff"
+  size="medium"
+  type="ball-zig-zag"
+  [fullScreen]="true"
+>
+  <p class="text-white">
+    {{ 'Loading...' | translate }}
+    @if (fileUpload?.files?.length > 0) { ({{ nUploadedFiles }} /
+    {{ fileUpload.files.length }}) }
```
Comment (Garfield-fr) [inline projects/sonar/src/app/record/files/upload-files/upload-files.component.html]: formatting is not really good for reading.

File: projects/sonar/src/app/record/files/upload-files/upload-files.component.html [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+      [disabled]="reachMaxFileLimit"
+      #fileUpload
+      [auto]="true"
+      [multiple]="true"
+      name="uploadFiles"
+      [chooseLabel]="'Add file' | translate"
+      [customUpload]="true"
+      (uploadHandler)="uploadHandler($event, '')"
+      mode="advanced"
+      [maxFileSize]="500 * 1024 * 1024"
```
Comment (Garfield-fr) [inline projects/sonar/src/app/record/files/upload-files/upload-files.component.html]: same here

File: projects/sonar/src/app/record/files/upload-files/upload-files.component.html [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+      </ng-template>
+
+    </p-fileUpload>
+  </div>
+
+  <p-orderList
+    #orderList
+    [value]="files"
+    (onReorder)="reorder()"
+    header="{{ 'Uploaded Files' | translate }}"
```
Comment (Garfield-fr) [inline projects/sonar/src/app/record/files/upload-files/upload-files.component.html]: `[header]="'Uploaded Files' | translate"`

File: projects/sonar/src/app/record/files/upload-files/upload-files.component.html [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+    [filterBy]="files.length > 5? 'label': null"
+    [dragdrop]="false"
+    controlsPosition="right"
+    class="w-100 col-12"
+    metaKeySelection="true"
+    filterPlaceholder="{{ 'Filter by label' | translate }}"
```
Comment (Garfield-fr) [inline projects/sonar/src/app/record/files/upload-files/upload-files.component.html]: `[filterPlaceholder]="'Filter by label' | translate"`

File: projects/sonar/src/app/record/files/file-item/file-item.component.html [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+  <ng-template pTemplate="header">
+    <ng-container
+      [ngTemplateOutlet]="fileLink"
+      [ngTemplateOutletContext]="{ file: file(), name: file().label }"
+    ></ng-container>
+  </ng-template>
+
+  <ng-template pTemplate="icons">
+    <button
+      title="{{ 'remove' | translate }}"
```
Comment (Garfield-fr) [inline projects/sonar/src/app/record/files/file-item/file-item.component.html]: `[title]="'remove' | translate"`

File: projects/sonar/src/app/record/files/upload-files/upload-files.component.html [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+      </ng-template>
+
+      <ng-template let-file pTemplate="file"> </ng-template>
+
+      <ng-template let-file pTemplate="content">
+        @if (reachMaxFileLimit) {
+        <div class="alert alert-danger" role="alert" translate>
+          The maximum number of files is reached. No additional upload is
+          allowed.
+        </div>
```
Comment (Garfield-fr) [inline projects/sonar/src/app/record/files/upload-files/upload-files.component.html]: Indentation

File: projects/sonar/src/app/record/files/upload-files/upload-files.component.html [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+        } @else {
+        <span translate>Drag and drop files.</span>
```
Comment (Garfield-fr) [inline projects/sonar/src/app/record/files/upload-files/upload-files.component.html]: Indentation

**Review by Garfield-fr — APPROVED** [review]
If you're not doing any tests, I think it's a good idea to delete the tests file.

File: package.json [hunk 9]
```diff
@@ -42,10 +42,10 @@
     "@angular/platform-browser-dynamic": "^17.3.10",
     "@angular/router": "^17.3.10",
     "@biesbjerg/ngx-translate-extract-marker": "^1.0.0",
-    "@ngx-formly/core": "^6.3.3",
-    "@ngx-formly/primeng": "^6.3.3",
+    "@ngx-formly/core": "6.3.2",
```
Comment (PascalRepond) [inline package.json:45]: v6.3.6 is available and fixes some issues, maybe good to test

**Review by PascalRepond — APPROVED** [review]

**Pull request merged**